src/context-menus/users/ban.ts

```diff
@@ -0,0 +1,78 @@
+import {
+	ActionRowBuilder,
+	ApplicationCommandType,
+	ContextMenuCommandBuilder,
+	ModalBuilder,
+	TextInputBuilder,
+	TextInputStyle
+} from 'discord.js';
+import { banHandler } from '@/commands/ban';
+import type { UserContextMenuCommandInteraction, ModalActionRowComponentBuilder} from 'discord.js';
+
+export async function banContextMenuHandler(interaction: UserContextMenuCommandInteraction): Promise<void> {
+	const target = interaction.targetUser;
+
+	const reasonInput = new TextInputBuilder()
+		.setCustomId('banReason')
+		.setLabel('Reason')
+		.setRequired(true)
+		.setStyle(TextInputStyle.Short);
+
+	const deleteMessagesInput = new TextInputBuilder()
+		.setCustomId('deleteMessages')
+		.setLabel('Delete recent message history (hours)')
+		.setPlaceholder('Leave blank to not delete any messages')
+		.setRequired(false)
+		.setStyle(TextInputStyle.Short);
+
+	const reasonActionRow = new ActionRowBuilder<ModalActionRowComponentBuilder>()
+		.addComponents(reasonInput);
+
+	const deleteMessagesActionRow = new ActionRowBuilder<ModalActionRowComponentBuilder>()
+		.addComponents(deleteMessagesInput);
+
+	const banModal = new ModalBuilder()
+		.setCustomId('banModal')
+		.setTitle(`Ban ${target.tag}`)
+		.addComponents(reasonActionRow, deleteMessagesActionRow);
+
+	await interaction.showModal(banModal);
+
+	let modalSubmitInteraction;
+	try {
+		modalSubmitInteraction = await interaction.awaitModalSubmit({
+			time: 5 * 60 * 1000,
```

Comment:
Is 5 minutes long enough? Not sure how much detail people go into

Reply:
This could be increased. The only thing is that slightly less than 15 minutes would be the maximum time, as interaction tokens are only valid for 15 minutes.

Reply:
As long as the mods are happy with the timer I think it's all good

Reply:
I'm all good with increasing the timer to 14 minutes 30 seconds, to be safe.